Removes pragmas from CBMC-proof harnesses

feliperodri opened the pull request:
Signed-off-by: Felipe R. Monteiro <felisous@amazon.com>

### Resolved issues:

This PR is the first clean up necessary to address https://github.com/aws/s2n-tls/issues/2774. We will add an appropriate check for these harnesses subsequently. 

### Description of changes: 
N/A.

### Call-outs:
N/A.

### Testing:
N/A.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

SaswatPadhi reviewed (changes requested):
The changes LGTM. Two meta comments:

1. We don't seem to be using `old_array`, `old_set` etc. any more -- can we remove all of them as well?

2. Are you also planning to add checks to check if the array (or set) was indeed deallocated? If you're planning to do that in a separate change, can we leave #2774 open till then (i.e. remove `Resolves #2774.` from the PR description)?

feliperodri commented:
> The changes LGTM. Two meta comments:
> 
>     1. We don't seem to be using `old_array`, `old_set` etc. any more -- can we remove all of them as well?

I removed all unused `old_*` variables.

>     2. Are you also planning to add checks to check if the array (or set) was indeed deallocated? If you're planning to do that in a separate change, can we leave #2774 open till then (i.e. remove `Resolves #2774.` from the PR description)?

I updated the PR description so #2774 won't be closed after we merge this one.

SaswatPadhi reviewed (approved):
Thank you, LTGM.

camshaft reviewed (approved):
Nice!
